### ISSUE TITLE
Slim down Intel image

### ISSUE THREAD

**Issue by tobiasleibner** [dune-community/Dockerfiles#99]
Currently, the ``icc`` image installs ``intel-basekit`` and ``intel-hpckit``. The basekit packages alone [sum up to about 14 GB](https://github.com/dune-community/Dockerfiles/pull/82#issuecomment-682008537). However, I just learned that you can also install each package separately. The list of available packages can be obtained by
```bash
sudo -E apt-cache pkgnames intel | grep intel-oneapi | grep -v intel-oneapi-runtime
```
(see [Intel documentation](https://software.intel.com/content/www/us/en/develop/articles/installing-intel-oneapi-toolkits-via-apt.html), "List standalone components"). We probably only need the compilers, TBB, and maybe MPI and Python. This hopefully also reduces the number of [env variables](https://github.com/dune-community/Dockerfiles/blob/b15a81813a5d2e0ba1e1c4f826461f65e799c497/debian/include/full-intel#L15-L66) that we have to set in the image significantly.

**Comment by renefritze** [comment]
I don't think we _need_ intel python. It would require additional configuration in our config.opts to be used and I see no upside, unless someone wants to explore a potential performance gain over the default cpython.
If we're currently building with MPI enabled in CI, I would leave Intel MPI in the image. From my experience icc sometimes has problems with openmpi compiled with gcc.
